File: pod_api/bin/app/models/podcast_feed.dart
import 'package:webfeed/webfeed.dart';
import 'podcast_item.dart';

class PodcastFeed {
  final String title;
  final String lastBuild;
  final String description;
  final String image;
  final String owner;
  final List<PodcastItem> items;

  PodcastFeed({
    this.title,
    this.lastBuild,
    this.description,
    this.image,
    this.owner,
    this.items,
  });

  static PodcastFeed fromXML(String xmlString) {
    final rss = RssFeed.parse(xmlString);
    return PodcastFeed(
        title: rss.title,
        lastBuild: rss.lastBuildDate,
        description: rss.description,
        image: rss.image.url,
        owner: rss.copyright,
        items: rss.items.map((item) => PodcastItem.from(item)).toList());
  }

  Map<String, dynamic> toMap() {
    return {
      'title': title,
      'lastBuild': lastBuild,
      'description': description,
      'image': image,
      'owner': owner,
      'items': items.map((item) => item.toMap()).toList(),
    };
  }
}

File: podapi_ts/src/Models/PodcastFeed.ts
import PodcastFeedDTO from "./DTOs/PodcastFeedDTO";
import Model from "./Model";
import { IPodcastItem, PodcastItem } from "./PodcastItem";

// tslint:disable-next-line:interface-name
export interface IPodcastFeed {
  title: string;
  lastBuild: string;
  description: string;
  image: string;
  owner: string;
  items: IPodcastItem[];
}

export class PodcastFeed extends Model<PodcastFeedDTO> implements IPodcastFeed {
  public title: string;
  public lastBuild: string;
  public description: string;
  public image: string;
  public owner: string;
  public items: PodcastItem[];

  constructor(dto: PodcastFeedDTO) {
    super(dto);
  }

  public mapFromDTO(dto: PodcastFeedDTO): void {
    const channelValue = dto.rss.channel;
    this.title = channelValue.title;
    this.lastBuild = channelValue.lastBuildDate;
    this.description = channelValue.description;
    this.image = channelValue.image.url;
    this.owner = channelValue.owner.name;
    this.items = channelValue.item.map(item => new PodcastItem(item));
  }
}
